spark/src/test/java/org/apache/iceberg/spark/data/TestSparkOrcUnions.java

```diff
@@ -525,6 +531,87 @@ public void testDeepNestedSingleTypeUnion() throws IOException {
     }
   }
 
+  @Test
```

Comment:
Just for my learning:
this new test case seems to cover Issue 2 in the PR, do we also need additional unit test for testing single lvl union with partial projection for Issue 1?

Reply:
I think there's previous unit test already testing the single lv union.